File: 프로그래머스/0/120890. 가까운 수/가까운 수.js
function solution(array, n) {
    array.sort((x,y) => x - y)
    let differ = Infinity;
    let answer = 0;
    for(let i of array) { 
        if (Math.abs(n - i) < differ) {  
        differ = Math.abs(n - i) 
        answer = i 
        }
    }
    return answer; 
}

File: 프로그래머스/0/120890. 가까운 수/가까운 수.py
import math

def solution(array, n):
    array.sort()

    differ = float('inf') 
    answer = 0
    
    for i in array:
        current_diff = abs(n - i)
        if current_diff < differ:
            differ = current_diff
            answer = i
        elif current_diff == differ and i < answer:
            answer = i
    return answer